chainer/functions/noise/gaussian.py

```diff
@@ -28,23 +35,24 @@ def check_type_forward(self, in_types):
         )
 
     def forward_cpu(self, inputs):
-        self.retain_inputs(())
+        self.retain_inputs((1,))
 
         mean, ln_var = inputs
         if self.eps is None:
-            self.eps = numpy.random.standard_normal(ln_var.shape) \
-                                   .astype(numpy.float32)
+            self.eps = (
+                numpy.random.standard_normal(ln_var.shape)
+                .astype(numpy.float32)
+            )
 
         self.noise = numpy.exp(ln_var * mean.dtype.type(0.5)) * self.eps
         return utils.force_array(mean + self.noise),
 
     def forward_gpu(self, inputs):
-        self.retain_inputs(())
+        self.retain_inputs((1,))
 
-        cupy = cuda.cupy
         mean, ln_var = inputs
         if self.eps is None:
-            self.eps = cupy.random.standard_normal(
+            self.eps = cuda.cupy.random.standard_normal(
                 ln_var.shape, dtype=mean.dtype)
 
         self.noise = cuda.cupy.empty_like(mean)
```

Comment:
`self.noise` is enough to be a local variable now.

Reply:
Ditto!